reascripts/common/libs/PathUtil.lua

```diff
@@ -0,0 +1,91 @@
+--[[
+
+  PathUtil.lua - Path utilities for ReaScript
+
+]]--
+
+PathUtil = {}
+
+-- Apply an extension to a file path, if it doesn't already have one.
+-- Empty or nil path returns the empty string.
+PathUtil.apply_extension = function(filepath, extension)
+  if not filepath or #filepath < 1 then
+    return ""
+  end
+
+  local path = ""
+
+  local path_part, file_part = filepath:match("(.+)[\\/](.+)")
+
+  if path_part then
+    path = path_part .. PathUtil._path_separator()
+    filepath = file_part
+  end
+
+  -- split the path by ".", if last chunk is alphanumeric don't apply extension
+  local chunks = {}
+  for chunk in filepath:gmatch("[^%.]+") do
+    table.insert(chunks, chunk)
+  end
+
+  local last_chunk = chunks[#chunks]
+
+  if #chunks > 1 and last_chunk:find("%w+") then
+    return path .. filepath
+  end
+
+  return path .. filepath .. '.' .. extension
+end
+
+-- Returns the given path if a full path, otherwise returns given path
+-- relative to the REAPER project resource directory.
+PathUtil.get_real_path = function(path_arg)
+  if PathUtil._is_full_path(path_arg) then
+    return path_arg
+  end
+
+  local project_path = reaper.GetProjectPath()
+  return project_path .. PathUtil._path_separator() .. path_arg
+end
+
+-- Returns the command to reveal the given path in the OS file explorer.
+-- Works for Windows and macOS. To quote a wise philosopher, "Linux is a mess."
```

Comment:
LOL. So, there is a utility that is pretty standard on Linux called xdg-open. I can help test if you want.

Reply:
i noticed that while i was searching around, but it seemed like there was no guarantee that the command would exist. i think i didn't feel ready to introduce a layer of checking for that (unlike the mac/windows side where i could be reasonably sure that those commands were built in). 🤔

Reply:
Yeah, that's accurate. The command should exist for users of common desktop environments like GNOME and KDE. There's also a small possibility that the wrong program will open; I've seen threads about directories opening in Firefox, VSCode, and other things besides the expected Nautilus (or similar).